radix-engine/src/system/system_callback.rs

```diff
@@ -61,6 +64,158 @@ pub type SystemBootSubstate = SystemBoot;
+                        .auth_zone_init
+                        .simulate_every_proof_under_resources
+                        .clone();
+                    let virtual_non_fungibles =
+                        intent.auth_zone_init.initial_non_fungible_id_proofs.clone();
+                    let auth_zone = AuthModule::create_auth_zone(
+                        &mut system_service,
+                        None,
+                        virtual_resources,
+                        virtual_non_fungibles,
+                    )?;
+
+                    api.kernel_set_call_frame_data(Actor::Function(FunctionActor {
+                        blueprint_id: BlueprintId::new(
+                            &TRANSACTION_PROCESSOR_PACKAGE,
+                            TRANSACTION_PROCESSOR_BLUEPRINT,
+                        ),
+                        ident: TRANSACTION_PROCESSOR_RUN_IDENT.to_string(),
+                        auth_zone,
+                    }))?;
+                }
+
+                {
+                    let mut system_service = SystemService::new(api);
+                    let intent = intents.get(0).unwrap();
+
+                    let txn_processor = TxnProcessor::<InstructionV1>::init(
+                        intent.encoded_instructions.clone(),
+                        global_address_reservations.clone(),
+                        intent.blobs.clone(),
+                        MAX_TOTAL_BLOB_SIZE_PER_INVOCATION,
+                        &mut system_service,
+                    )?;
+                    let output = txn_processor.execute(&mut system_service)?;
+
+                    let owned_nodes = api.kernel_get_owned_nodes()?;
+                    System::auto_drop(owned_nodes, api)?;
+
+                    let actor = api.kernel_get_system_state().current_call_frame;
+                    match actor {
+                        Actor::Function(FunctionActor { auth_zone, .. }) => {
+                            let auth_zone = auth_zone.clone();
+                            let mut system_service = SystemService::new(api);
+                            AuthModule::teardown_auth_zone(&mut system_service, auth_zone)?;
+                        }
+                        _ => {
+                            panic!("unexpected");
+                        }
+                    }
+
+                    let owned_nodes = api.kernel_get_owned_nodes()?;
+                    if !owned_nodes.is_empty() {
+                        return Err(RuntimeError::KernelError(KernelError::OrphanedNodes(
+                            owned_nodes,
+                        )));
+                    }
+
+                    output
+                }
+            }
+        };
+
+        Ok(output)
+    }
+
+    pub fn should_consume_cost_units<Y: SystemBasedKernelApi>(&self, api: &mut Y) -> bool {
+        match self {
+            VersionedSystemLogic::V1 => {
+                // Skip client-side costing requested by TransactionProcessor
```

Comment:
Out of interest, do you know why we skip this particular costing for the transaction processor?

Maybe it's absorbed by the flat transaction cost?